### ISSUE TITLE
Support pre-calculated affinity matrix in ht.cluster.Spectral

### ISSUE THREAD

**Issue by ClaudiaComito** [helmholtz-analytics/heat#1740]
**Feature functionality**
Allow users to provide a pre-calculated affinity matrix when computing spectral clustering, as in [scikit-learn SpectralClustering](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.SpectralClustering.html).

Also generally adapt the `ht.Spectral` API to scikit-learn.

**Additional context**
Related to parallelization efforts within the [SCIMES](https://github.com/Astroua/SCIMES) project with @dcolombo

**Comment by ClaudiaComito** [comment]
eigensolver based on Lanczos at the moment, update within the ESAPCA program

**Comment by ClaudiaComito** [comment]
Hi @dcolombo,

I've updated Spectral on the branch features/1740-etc. After a git pull you should have:

 - An API that's a bit more similar to scikit-learn (`SpectralClustering` instead of `Spectral`, `affinity` instead of `metric`)
 - the option to set `affinity = 'precomputed'`

Keep in mind that the matrix decomposition on Heat is currently done with the Lanczos method, so you'll definitely get different results wrt scikit-learn.
